airbyte-cdk/python/airbyte_cdk/sources/declarative/schema/default_schema_loader.py

```diff
@@ -23,9 +23,13 @@ class DefaultSchemaLoader(SchemaLoader, JsonSchemaMixin):
     """
 
     config: Config
+    name: InitVar[str]
     options: InitVar[Mapping[str, Any]]
 
-    def __post_init__(self, options: Mapping[str, Any]):
+    def __post_init__(self, name: str, options: Mapping[str, Any]):
+        options = options or {}
+        if "name" not in options:
+            options["name"] = name
```

Comment:
If name is defined, why not use `self.name` directly at https://github.com/airbytehq/airbyte/pull/21516/files#diff-eebfee995e9a963a52d517c0ebcab9579872dee179cf961719740e6c3fdddc9fR48 instead of `self._options.get("name", "")` that was generating the error? Do we need the options at all?

Reply:
`name` isn't available on `self` here, and at least for now for consistency with the rest of the factory we do want to override `model.name` if `options` is provided and includes it. Removing `options` is a possible refactor but felt out of scope for this PR (we've talked about doing a larger effort to remove it where it's not needed as there are various places where that's the case).